pkg/operator/k8sutil/prometheus.go

```diff
@@ -80,6 +80,7 @@ func CreateOrUpdateServiceMonitor(serviceMonitorDefinition *monitoringv1.Service
 		return nil, fmt.Errorf("failed to retrieve servicemonitor. %v", err)
 	}
 	oldSm.Spec = serviceMonitorDefinition.Spec
+	oldSm.ObjectMeta.Labels = serviceMonitorDefinition.ObjectMeta.Labels
```

Comment:
Do we need to do the same for annotations?

Reply:
I actually had that in there at first, but iirc there is currently no way to set annotations on these resources via rook CRDs and the templates do not contain any, too - so I removed that, for not cluttering code with seemingly unneeded things.

Reply:
I see, we should probably add the annotations in a future PR, but looks good for now!